contracts/lootbox/LootboxByItem.sol

```diff
@@ -116,9 +116,9 @@ contract LootboxByItem is ILootbox, ERC1155Upgradeable, AccessControlUpgradeable
             }
 
             // Don't double count guaranteed items.
-            if(rewards[i].probability < 1000000) {
+            if(rewards[i].probability < 1e6) {
                 uint256 randomVal = LibLootbox.random(msg.sender, seed);
```

Comment:
Not sure if it matters, but we can replace msg.sender with _msgSender() for all the lootbox contracts

Reply:
this is a good catch! _msgSender() is better than msg.sender